pkg/asset/rhcos/image.go

```diff
@@ -52,101 +55,107 @@ func (i *Image) Dependencies() []asset.Asset {
 func (i *Image) Generate(ctx context.Context, p asset.Parents) error {
 	if oi, ok := os.LookupEnv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); ok && oi != "" {
 		logrus.Warn("Found override for OS Image. Please be warned, this is not advised")
-		*i = Image(oi)
+		*i = *MakeAsset(oi)
```

Comment:
so in the override case - the same image would be used for cp/computes?

Reply:
Yes. If we need to support 2 image overrides, then we'd have to either require the `OPENSHIFT_INSTALL_OS_IMAGE` to have both images (e.g, separated by `,`) or we'd have to introduce a new env var for compute images.

Reply:
maybe  a change to the installer doc mentioning this is also needed

Reply:
We don't have docs for that env var as it's not officially supported.

Reply:
@Prashanth684 can you envision a situation where we'd need to override both controlplane and compute images with distinct values for multi-arch clusters? I'd rather not touch this env var as it's used by agent and powervs (at least not in this PR).

Reply:
> I'd rather not touch this env var as it's used by agent and powervs
it makes sense to address separately. Do they use it for installation - if so we have to make sure that it works or that we don't allow them to provision a cluster with multi-arch computes on day 0?

Reply:
agent: used when generating the base ISO: https://github.com/openshift/installer/blob/master/pkg/asset/agent/image/baseiso.go#L149-L151

powervs: it seems it's abusing the envvar: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/powervs/platform.go#L19-L24

> we don't allow them to provision a cluster with multi-arch computes on day 0

The only way to bypass the heterogeneous validation in the installer is by setting the feature gate: https://github.com/openshift/installer/pull/8698/commits/d9e5e6686f69ac523d65bddc80a5e07ba226c3c7

That only happens for AWS with this PR.